sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_region.c

```diff
@@ -106,6 +107,8 @@ uint32_t gnrc_lorawan_pick_channel(gnrc_lorawan_t *mac)
     for (int i = 0; i < pos + 1; i++) {
         state = bitarithm_test_and_clear(state, &index);
     }
+
+    mac->last_chan_idx = index;
```

Comment:
since this function only picks a random channel, I would be in favor of maintaining "mac->last_chan_idx" outside of this function. It makes it easier to maintain as well

Reply:
I am now managing it inside `gnrc_lorawan_mcps_request` by just looping through the channels and looking for the correct one. With this change I had removed the call to `gnrc_lorawan_pick_channel` from `gnrc_lorawan_event_retrans_timeout` which I had added because otherwise the `last_chan_idx` wont be correct.

I am wondering if this is bad since `gnrc_lorawan_event_retrans_timeout` might wan't to use a new channel ?

Reply:
> I am wondering if this is bad since gnrc_lorawan_event_retrans_timeout might wan't to use a new channel ?

Hmmmm yes, the retransmssion should follow the same rules of any transmission. So it should choose a different channel.

Reply:
Due to this I now moved the channel picking + the calculation of the uplink mic into the `_transmit_pkt` function. This is the cleanest solution I could find.